### PR TITLE
sanity: check /dev/loop-control device as part of sanity checks

### PR DESCRIPTION
**Pull request by stolowski**
Check if /dev/loop-control exists and has proper major/minor. This was prompted by today's investigation of a broken tumbelweed system where this device was missing.

### CODE REVIEW

**Comment by stolowski** [comment]
Hmm it fails on lxd tests:

```
 lxd.lxc exec my-ubuntu snap install test-snapd-tools
error: system does not fully support snapd: cannot stat "/dev/loop-control": no
       such file or directory
```

File: sanity/loop.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+var majorMinor = func(rdev int) (major, minor int) {
+	major = int(rdev / 256)
+	minor = int(rdev % 256)
+	return major, minor
+}
+
+func checkLoopControl() error {
+	var st syscall.Stat_t
+	err := syscall.Stat(loopControlPath, &st)
```
Comment (bboozzoo) [inline sanity/loop.go]: As discussed on IRC, this should depend on `squashfs.FsType()`, we only need loop when the actual `squashfs` is in use.

Maybe it'd make sense to move this check into the the squashfs one?

**Comment by stolowski** [comment]
Fails due to SELinux denials on Fedora, needs policy tweak, will investigate.

File: sanity/loop.go (hunk 1 above)
Comment (stolowski) [reply sanity/loop.go]: Done.

File: sanity/squashfs.go [hunk 2]
```diff
@@ -141,5 +141,9 @@ func checkSquashfsMount() error {
 		return fmt.Errorf("unexpected squashfs canary content: %q", content)
 	}
 
+	// lastly, verify that loop device is usable (which would not be caught by squashfs tests above)
```
Comment (pedronis) [inline sanity/squashfs.go]: the comment needs to be expended to explain why we care about this device even if apparently mount  as above works?  when does the device gets into the picture?

**Comment by stolowski** [comment]
Closing, I think the conclusions that led to this were premature. Indeed mount -t squashfs works even after `rm -rf /dev/loop-control`. Moreover, in my test /dev/loop-control gets restored on opensuse (where  the problem originally reported) after reboot. Thanks @pedronis for questioning this.

**Pull request closed without merging**